src/libraries/System.Private.CoreLib/src/System/Int16.cs

```diff
@@ -631,24 +631,11 @@ public static short Log2(short value)
         /// <inheritdoc cref="INumber{TSelf}.CopySign(TSelf, TSelf)" />
         public static short CopySign(short value, short sign)
         {
-            short absValue = value;
-
-            if (absValue < 0)
-            {
-                absValue = (short)(-absValue);
-            }
-
-            if (sign >= 0)
+            if (value == short.MinValue && sign >= 0)
             {
-                if (absValue < 0)
-                {
-                    Math.ThrowNegateTwosCompOverflow();
-                }
-
-                return absValue;
+                Math.ThrowNegateTwosCompOverflow();
             }
-
-            return (short)(-absValue);
+            return (short)(value * Math.SignZeroToOne(value ^ sign));
```

Comment:
Multiplication is much more expensive on older CPUs and may significantly regress performance.

It's not necessarily "cheap" on modern CPUs either (still a minimum of 3-4 cycles for 32-bit, and 3-6 cycles for 64-bit) and while reduction of branches can be goodness, it might hurt things overall and hinder JIT optimizations for some constants.